File: 351004/Zalivako/DistributedComputing/src/Core/Entities/Editor.cs
namespace Core.Entities
{
    public static class UserRoles
    {
        public const string Admin = "ADMIN";
        public const string Customer = "CUSTOMER";
    }

    public class Editor(
        string login,
        string password,
        string firstname,
        string lastname) : Entity
    {
        public string Login { get; set; } = login;

        public string Password { get; set; } = password;

        public string Firstname { get; set; } = firstname;

        public string Lastname { get; set; } = lastname;

        public string Role { get; set; } = UserRoles.Customer;

        public IEnumerable<News>? News { get; set; }
    }
}

File: 351001/shaminko/module-publisher/src/main/java/by/shaminko/distcomp/entities/Editor.java
package by.shaminko.distcomp.entities;

import jakarta.persistence.*;
import jakarta.validation.constraints.Size;
import lombok.*;

@Entity
@Getter
@Setter
@ToString
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "tbl_creator")
public class Editor {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    long id;

    @Column(unique = true)
    @Size(min = 2, max = 64)
    String login;

    @Size(min = 8, max = 128)
    String password;

    @Size(min = 2, max = 64)
    String firstname;

    @Size(min = 2, max = 64)
    String lastname;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 16)
    UserRole role;
}
